pkg/operator/staticpod/certsyncpod/certsync_cmd.go

```diff
@@ -55,6 +57,18 @@ func (o *CertSyncControllerOptions) Complete() (*CertSyncController, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// When the kubeconfig content change, commit suicide to reload its content.
+	observer, err := fileobserver.NewObserver(500 * time.Second)
+	if err != nil {
+		return nil, err
+	}
+	initialContent, _ := ioutil.ReadFile(o.KubeConfigFile)
+	observer.AddReactor(fileobserver.ExitOnChangeReactor, map[string][]byte{o.KubeConfigFile: initialContent}, o.KubeConfigFile)
+
+	stopCh := make(chan struct{})
+	go observer.Run(stopCh)
```

Comment:
side effect in a Complete func looks fishy. Move it into a Start or Run method.

Reply:
sure